### PR TITLE
Index templates: collect JavaScript errors.

### PR DESCRIPTION
**Pull request by frankban**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4768/
Test PASSed.

File: jujugui/templates/index.html.mako [hunk 1]
```diff
@@ -37,6 +37,14 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
     <meta name="viewport" content="width=device-width,initial-scale=1.0 maximum-scale=1.0, user-scalable=no">
     <meta name="description" content="">
     <meta name="author" content="Juju team">
+    <script type="text/javascript">
+      // This is used as part of Juju GUI automated QA.
+      window.jsErrors = [];
+      window.onerror = function(msg, source, line) {
+        window.jsErrors[window.jsErrors.length] = msg + ' (' + source + ' at line ' + line + ')';
```
Comment (hatched) [inline jujugui/templates/index.html.mako]: window.jsErrors.push(...) instead of using the length.

**Comment by hatched** [comment]
:+1: With that trivial change!

Thanks, this will be helpful with ci

File: jujugui/templates/index.html.mako (hunk 1 above)
Comment (frankban) [inline jujugui/templates/index.html.mako]: Done.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4770/
Test FAILed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4771/
Test FAILed.

**Comment by frankban** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4773/
Test PASSed.

**Pull request merged**